app/Services/DiscourseService.php

```diff
@@ -230,30 +230,72 @@ public function syncUsersToGroups($idgroups = NULL) {
+                    // We need to update it.  To do that, we first have to upload the image file to Discourse.
+                    Log::debug("Need to update flair_url with ". $group->groupImagePath());
+
+                    // Upload an image.  We need the MIME type of the file.
+                    $fh = fopen(public_path('uploads/mid_' . $group->groupImage->image->path),'r');
+
+                    $data = [
+                        [
+                            'name' => 'upload_type',
+                            'contents' => 'group_flair'
+                        ],
+                        [
+                            'name' => 'type',
+                            'contents' => mime_content_type($fh)
+                        ],
+                        [
+                            'name' => 'sha1_checksum',
+                            'contents' => sha1_file($group->groupImagePath())
+                        ],
+                        [
+                            'name' => 'file',
+                            'contents' => file_get_contents($group->groupImagePath()),
+                            'filename' => 'GroupLogo' . $idgroups
+                        ]
+                    ];
+
+                    $response = $client->request('POST', '/uploads.json', [
+                        'multipart' => $data
+                    ]);
+
+                    Log::debug("Response from upload ". $response->getStatusCode() . " " . $response->getBody());
+
+                    if ($response->getStatusCode() === 200) {
+                        // Now we can update the group to use it.
+                        $rsp = json_decode($response->getBody(), TRUE);
+
+                        if ($rsp && array_key_exists('id', $rsp)) {
+                            $uploadId = $rsp['id'];
+                            $response = $client->request('PUT', "/g/$discourseId.json", [
+                                'form_params' => [
+                                    'group' => [
+                                        'flair_upload_id' => $uploadId
+                                    ]
+                                ]
+                            ]);
+
+                            Log::debug("Response from flair_url update " . $response->getBody());
+
+                            if ($response->getStatusCode() === 200) {
+                                // Update the group to record that we've uploaded, then we'll skip this next time
+                                // through.
+                                Log::debug("Updated flair url OK for {$discourseId} to {$group->groupimage->idimages}");
+                                $group->discourse_logo = $group->groupimage->image->idimages;
+                                $group->save();
+                                Log::debug("...saved update to group");
+                            } else {
+                                Log::error("Failed to update flair url for {$discourseId}");
```

Comment:
Wondering if we'll end up with lots of errors in Sentry if this fails - I can't remember if we found a way to log argument values in Sentry outside of the title?

Reply:
As per JIRA I think at least some flair images have been set up manually.

Good point about Sentry..  I'll remove the ID for now.

You can add "context" to Sentry events, but it's a bit more of a faff.  Generally that time is better spent fixing them :-).